kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/kylin/engine/spark/builder/CubeTableEncoder.scala

```diff
@@ -63,7 +67,20 @@ object CubeTableEncoder extends Logging {
           .select(columns: _*)
       }
     )
+
     ds.sparkSession.sparkContext.setJobDescription(null)
+
+    //repartition by a single column during dict encode step before is more easily to cause data skew, add step to void such case.
+    if (!cols.isEmpty && seg.kylinconf.rePartitionEncodedDatasetWithRowKey) {
+      val colsInDS = partitionedDs.schema.map(_.name)
+      val rowKeyColRefs = seg.allRowKeyCols.map(colDesc => convertFromDot(colDesc.identity)).filter(colsInDS.contains).map(col)
+      //if not set in config, use the largest partition num during dict encode step
+      if (seg.kylinconf.getRepartitionNumAfterEncode < 1) {
```

Comment:
This is wrong, needs to modify to `seg.kylinconf.getRepartitionNumAfterEncode > 0`, otherwise when set this parameter to 5, for example, it doesn't use the value of this parameter to set to `repartitionSizeAfterEncode`.

Reply:
haha, misled by you, fixed ~